### PR TITLE
feature/web-3814: application creation with commands

### PR DESCRIPTION
**Pull request by fredmajor**
This adds Django Commands to create/update OAuth Applications in both client and provider.

- oauth2_app_maker - abstract parent command to create/update an oauth application instance in DB
- oauth2client_app - child command to create/update client's applications
- oauth2provider_app - child command to create/update provider's applications
- LoggingBaseCommand - add standard python logger to django's commands. Set log level from `--verbosity` argument

Follow up work, to be checked-in alongside JWT grant support:
- rename updated_at to updated on oauth2_client.models.Application. The `oauth2client_app` command isn't functional without that change.
- replace Application model used for testing with the in-house one. To remove external test dependency. [here](http://github.com/Livit/Labster.OAuth2Client/blob/4ac9c5d9da1f41502d16fc9791f68b94f139e19b/oauth2_client/tests/test_oauth2_app_maker_cmd.py#L44-L44)
- enable tests that depend on JWT grant support, [here](http://github.com/Livit/Labster.OAuth2Client/blob/4ac9c5d9da1f41502d16fc9791f68b94f139e19b/oauth2_client/tests/test_oauth2client_app_cmd.py#L25-L25) Re-working them only for formal sake would be unjustified effort, as the work already has been done, but doesn't fit this PR.

Full documentation: https://liv-it.atlassian.net/wiki/spaces/WEB/pages/891191387/OAuth2Client+documentation

Related Jiras:
https://liv-it.atlassian.net/browse/WEB-3814?atlOrigin=eyJpIjoiZDcwNGE2ZWRhNGEwNGQxZDgxZTA5MzY0M2I4NDU2NTEiLCJwIjoiaiJ9

### CODE REVIEW

File: oauth2_client/tests/test_oauth2client_app_cmd.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+"""
```
Comment (auraz) [inline oauth2_client/tests/test_oauth2client_app_cmd.py]: You may not need _app_ in these filenames - it reads well without it too.

File: oauth2_client/management/commands/oauth2provider_app.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+"""
+Create an instance of oauth2_provider's Application in DB, from CLI.
+"""
+from oauth2_client.management.commands.oauth2_app_maker import Command as ParentCommand
```
Comment (auraz) [inline oauth2_client/management/commands/oauth2provider_app.py]: Please change name from Command to something more specific as it can be confused with django command.

File: oauth2_client/management/commands/oauth2client_app.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+"""
+Create an instance of OAuth2Client's Application in DB, from CLI.
+"""
+import json
+
+from oauth2_client.management.commands.oauth2_app_maker import Command as ParentCommand
+from oauth2_client.models import Application as AppModel
```
Comment (auraz) [inline oauth2_client/management/commands/oauth2client_app.py]: Why change name?

File: oauth2_client/management/commands/oauth2provider_app.py (hunk 2 above)
Comment (fredmajor) [reply oauth2_client/management/commands/oauth2provider_app.py]: `oauth2_app_maker` **is** a django command (it extends django's BaseCommand, via LoggingBaseCommand). Django requires every command class to be named `Command`, otherwise it won't be picked up by the framework and registered as a command. I want the `oauth2_app_maker` to be picked up by the framework (even though it is abstract) because that allows me to test it better. I tried to explain that in [this comment](http://github.com/Livit/Labster.OAuth2Client/blob/4ac9c5d9da1f41502d16fc9791f68b94f139e19b/oauth2_client/tests/test_oauth2_app_maker_cmd.py#L4-L4).

To summarize: renaming the parent command to some different name would result in (slightly!) less comprehensive tests. Please make the final call @auraz

File: oauth2_client/management/commands/oauth2client_app.py (hunk 3 above)
Comment (fredmajor) [reply oauth2_client/management/commands/oauth2client_app.py]: no reason, removed

File: oauth2_client/tests/test_oauth2client_app_cmd.py (hunk 1 above)
Comment (fredmajor) [reply oauth2_client/tests/test_oauth2client_app_cmd.py]: you mean only test files or the actual commands too? Please bear in mind that renaming files that contain commands will imply renaming the commands themselves too. Command rename would look like that:

oauth2client_app -> oauth2client
oauth2provider_app -> oauth2provider

I don't have enough experience with django naming conventions, I would be really grateful if you can make the call @auraz

File: oauth2_client/management/commands/oauth2provider_app.py (hunk 2 above)
Comment (auraz) [reply oauth2_client/management/commands/oauth2provider_app.py]: Sounds OK, good then!

File: oauth2_client/tests/test_oauth2client_app_cmd.py (hunk 1 above)
Comment (auraz) [reply oauth2_client/tests/test_oauth2client_app_cmd.py]: Only test, sorry for the confusion.

Comment (auraz) [reply oauth2_client/tests/test_oauth2client_app_cmd.py]: Approved anyway as it is very minor.
